### PR TITLE
Create macro for selinux ansible/bash remediation.

### PR DESCRIPTION
**Pull request by ggbecker**
#### Description:

Create macro for selinux ansible/bash remediation.

Affected rules:
 - selinux_policytype
 - selinux_state

#### Rationale:

- Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1750526
- #4838

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule selinux_policytype:
 Templatazation usage changed.
 Templatization usage changed.
 Found change in bash remediation.
Rule selinux_state:
 Templatazation usage changed.
 Templatization usage changed.
 Found change in bash remediation.
Macro ansible_selinux_config_set:
 In Ansible remediation for selinux_policytype.
 In Ansible remediation for selinux_state.
Macro bash_coredump_config_set:
 In Bash remediation for coredump_disable_backtraces.
 In Bash remediation for coredump_disable_storage.
Macro bash_selinux_config_set:
 In Bash remediation for selinux_policytype.
 In Bash remediation for selinux_state.

**Recommended tests to execute:**
 build_product ol8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol8-ds.xml selinux_policytype
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml selinux_policytype
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol8-ds.xml selinux_state
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml selinux_state

File: linux_os/guide/system/selinux/selinux_policytype/tests/selinuxtype_minimum.fail.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+# profiles = xccdf_org.ssgproject.content_profile_C2S, xccdf_org.ssgproject.content_profile_ospp
+
+SELINUX_FILE='/etc/selinux/config'
+
+if grep -s '^[[:space:]]*SELINUXTYPE' $SELINUX_FILE; then
+	sed -i 's/^\([[:space:]]*SELINUXTYPE[[:space:]]*=[[:space:]]*\).*/\minimum/' $SELINUX_FILE
+else
+	echo 'SELINUXTYPE=minimum' >> $SELINUX_FILE
+fi
```
Comment (yuumasato) [inline linux_os/guide/system/selinux/selinux_policytype/tests/selinuxtype_minimum.fail.sh:10]: The test is fine.

But as a side note, I think in general the test scenarios can be more direct and destructive.
Like for example, just remove the whole file and add the line subject of the test.
Thoughts on this?

**Comment by yuumasato** [comment]
The remediations require a reboot too, :)

File: linux_os/guide/system/selinux/selinux_policytype/tests/selinuxtype_minimum.fail.sh (hunk 1 above)
Comment (matusmarhefka) [reply linux_os/guide/system/selinux/selinux_policytype/tests/selinuxtype_minimum.fail.sh:10]: Agree, btw the sed command is wrong, should be `../\1minimum/`.

Comment (ggbecker) [reply linux_os/guide/system/selinux/selinux_policytype/tests/selinuxtype_minimum.fail.sh:10]: `../\1minimum/` will fix.

Agree with the test scenario approach. But I followed the pattern of other tests.

But maybe, sometimes this destructive approach can be dangerous, for example... if we delete a file that contain other configurations that are required to make the system work then we may end with an inaccessible system.

Funny thing, I was testing if the selinux configuration was case insensitive and for one case it was invalid and I couldn't access the system and it took a while to figure it out ;( .

I would rather keep the test as it is right now. But for future ones I can consider this way of thinking.

File: shared/macros-ansible.jinja [hunk 2]
```diff
@@ -217,6 +217,17 @@ value: "Setting={{ varname1 }}"
...
 
+{{#
+  High level macro to set a parameter in /etc/selinux/config.
+  Parameters:
+  - msg: the name for the Ansible task
+  - parameter: parameter to be set in the configuration file
+  - value: value of the parameter
+#}}
+{{%- macro ansible_selinux_config_set(msg='', parameter='', value='') %}}
+{{{ ansible_set_config_file(msg, "/etc/selinux/config", parameter=parameter, value=value, create="no", separator="=", separator_regex="\s*=\s*") }}}
```
[the remark this answers is not in the document]
Comment (ggbecker) [reply shared/macros-ansible.jinja:228]: ok

File: shared/macros-bash.jinja [hunk 3]
```diff
@@ -86,6 +86,21 @@ populate {{{ name }}}
...
+{{{ set_config_file(
+        path="/etc/selinux/config",
+        parameter=parameter,
+        value=value,
+        create=true,
+        insert_after="",
+        insert_before="",
+        insensitive=true,
+        separator="=",
+        separator_regex="\s*=\s*",
```
[the remark this answers is not in the document]
Comment (ggbecker) [reply shared/macros-bash.jinja:99]: Ok, will update.

File: linux_os/guide/system/selinux/selinux_policytype/tests/selinuxtype_minimum.fail.sh (hunk 1 above)
Comment (ggbecker) [reply linux_os/guide/system/selinux/selinux_policytype/tests/selinuxtype_minimum.fail.sh:10]: Kids, don't this at home:

`sed -i 's/^\([[:space:]]*SELINUXTYPE[[:space:]]*=[[:space:]]*\).*/\1taRgEteD/' /etc/selinux/config` 
```
         Starting Switch Root...
[    2.821988] systemd-journald[207]: Received SIGTERM from PID 1 (systemd).
[    2.841567] systemd: 13 output lines suppressed due to ratelimiting
[    2.856533] systemd[1]: Failed to load SELinux policy.
[!!!!!!] Failed to load SELinux policy, freezing.
[    2.861912] systemd[1]: Freezing execution.
[    3.182289] pciehp 0000:00:01.6:pcie004: Failed to check link status
```

Comment (ggbecker) [reply linux_os/guide/system/selinux/selinux_policytype/tests/selinuxtype_minimum.fail.sh:10]: When I fixed the `minimum` sed, the test scenario started to fail on my RHEL8.1 VM after a restart. I'll remove it.

**Comment by mildas** [comment]
**Changes identified:**
Rule selinux_policytype:
 Text changed in OVAL check.
 Templatazation usage changed.
 Templatization usage changed.
 Found change in bash remediation.
Rule selinux_state:
 Text changed in OVAL check.
 Templatazation usage changed.
 Templatization usage changed.
 Found change in bash remediation.
Macro ansible_selinux_config_set:
 In Ansible remediation for selinux_policytype.
 In Ansible remediation for selinux_state.
Macro bash_coredump_config_set:
 In Bash remediation for coredump_disable_storage.
 In Bash remediation for coredump_disable_backtraces.
Macro bash_selinux_config_set:
 In Bash remediation for selinux_policytype.
 In Bash remediation for selinux_state.

**Recommended tests to execute:**
 build_product ol8
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol8-ds.xml selinux_policytype
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml selinux_policytype
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol8-ds.xml selinux_state
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol8-ds.xml selinux_state

**Pull request merged**